### PR TITLE
Migrate the Docs to Docusaurus

### PR DESCRIPTION
**Pull request by saig0**
## Description

* replace the current docs with Docusaurus
* no changes on the content itself
* create a new landing page
* version the docs for 1.10, 1.11, 1.12, and 1.13 (develop/master)
* add a GitHub Actions workflow to build and deploy the docs

Preview:

image

```
cd docs
npm run start
```

## Related issues

none

### CODE REVIEW

**Comment by saig0** [comment]
@tasso94 / @koevskinikola please have a look at the PR. I changed the tool for the docs to Docusaurus and migrated the content. You don't need to go over all files. The content is the same, except for the new landing page. I just added metadata and adjusted the structure.

**Comment by saig0** [comment]
@tasso94 / @koevskinikola ping :bellhop_bell: Please have a look at the PR :)

**Comment by tasso94** [comment]
Hi @saig0,

Cool migration! 

Let us figure out in the team who could take this over. I'll come back to you asap.

Best,
Tassilo

**Comment by tasso94** [comment]
Hi @saig0,

I'll take care of the review.

Best,
Tassilo

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  },
+  presets: [
+    [
+      '@docusaurus/preset-classic',
+      {
+        docs: {
+          sidebarPath: require.resolve('./sidebars.js'),
+          // Please change this to your repo.
+          editUrl:
+            'https://github.com/facebook/docusaurus/edit/master/website/',
```
Comment (tasso94) [inline docs/docusaurus.config.js:73]:
```suggestion
            'https://github.com/camunda/feel-scala/edit/master/docs/',
```

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            current: {
+              label: `1.13 (unreleased)`,
+            },
+          },
+        },
+        blog: {
+          showReadingTime: true,
+          // Please change this to your repo.
+          editUrl:
+            'https://github.com/camunda/feel-scala/edit/master/new-docs/',
```
Comment (tasso94) [inline docs/docusaurus.config.js:86]:
```suggestion
            'https://github.com/camunda/feel-scala/edit/master/docs/',
```

File: docs/docs/reference/developer-guide/function-provider-spi.md [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
-### Implement a Function using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/docs/reference/developer-guide/function-provider-spi.md:24]:
```suggestion
<TabItem value="scala">

```

File: docs/docs/reference/developer-guide/value-mapper-spi.md [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
-### Implement a Value Mapper using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/docs/reference/developer-guide/value-mapper-spi.md:22]:
```suggestion
<TabItem value="scala">

```

File: docs/versioned_docs/version-1.10/reference/developer-guide/function-provider-spi.md [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
-### Implement a Function using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/versioned_docs/version-1.10/reference/developer-guide/function-provider-spi.md:24]:
```suggestion
<TabItem value="scala">

```

File: docs/versioned_docs/version-1.10/reference/developer-guide/value-mapper-spi.md [hunk 6]
```diff
@@ -1,17 +1,25 @@
...
-### Implement a Value Mapper using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/versioned_docs/version-1.10/reference/developer-guide/value-mapper-spi.md:22]:
```suggestion
<TabItem value="scala">

```

File: docs/versioned_docs/version-1.11/reference/developer-guide/function-provider-spi.md [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
-### Implement a Function using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/versioned_docs/version-1.11/reference/developer-guide/function-provider-spi.md:24]:
```suggestion
<TabItem value="scala">

```

File: docs/versioned_docs/version-1.11/reference/developer-guide/value-mapper-spi.md [hunk 8]
```diff
@@ -1,17 +1,25 @@
...
-### Implement a Value Mapper using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/versioned_docs/version-1.11/reference/developer-guide/value-mapper-spi.md:22]:
```suggestion
<TabItem value="scala">

```

File: docs/versioned_docs/version-1.12/reference/developer-guide/function-provider-spi.md [hunk 9]
```diff
@@ -1,19 +1,27 @@
...
-### Implement a Function using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/versioned_docs/version-1.12/reference/developer-guide/function-provider-spi.md:24]:
```suggestion
<TabItem value="scala">

```

File: docs/versioned_docs/version-1.12/reference/developer-guide/value-mapper-spi.md [hunk 10]
```diff
@@ -1,17 +1,25 @@
...
-### Implement a Value Mapper using Scala
-
+<Tabs
+  defaultValue="scala"
+  values={[
+    {label: 'Scala', value: 'scala'},
+    {label: 'Java', value: 'java'},
+  ]}>
+  
+<TabItem value="scala">
```
Comment (tasso94) [inline docs/versioned_docs/version-1.12/reference/developer-guide/value-mapper-spi.md:22]:
```suggestion
<TabItem value="scala">

```

File: docs/package.json [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    "serve": "docusaurus serve",
+    "clear": "docusaurus clear"
+  },
+  "dependencies": {
+    "@docusaurus/core": "2.0.0-alpha.69",
+    "@docusaurus/preset-classic": "2.0.0-alpha.69",
+    "@mdx-js/react": "^1.6.21",
+    "clsx": "^1.1.1",
+    "react": "^16.8.4",
+    "react-dom": "^16.8.4"
```
Comment (tasso94) [inline docs/package.json:20]:
```suggestion
    "@docusaurus/core": "2.0.0-alpha.70",
    "@docusaurus/preset-classic": "2.0.0-alpha.70",
    "@mdx-js/react": "^1.6.22",
    "clsx": "^1.1.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
```

**Review by tasso94 — APPROVED** [review]
LGTM! Nice work! I really like the new layout and structure. 
Can we use [prism](https://docusaurus.io/docs/en/next/doc-markdown#using-prism-as-additional-syntax-highlighter) to allow syntax highlighting for the Scala code snippets? 

Please find some minor improvements below.

**Comment by saig0** [comment]
I applied your suggestions and added the syntax highlighting for Scala. Thanks for the review :tada:

**Pull request merged**